scipy/signal/windows/_windows.py

```diff
@@ -1548,40 +1605,40 @@ def chebwin(M, at, sym=True):
                       "about 45 dB.",
                       stacklevel=2)
     if _len_guards(M):
-        return np.ones(M)
+        return xp.ones(M)
     M, needs_trunc = _extend(M, sym)
 
     # compute the parameter beta
     order = M - 1.0
-    beta = np.cosh(1.0 / order * np.arccosh(10 ** (np.abs(at) / 20.)))
-    k = np.r_[0:M] * 1.0
-    x = beta * np.cos(np.pi * k / M)
+    beta = xp.cosh(1.0 / order * xp.acosh(10 ** (abs(at) / 20.)))
+    k = xp.arange(M) * 1.0
+    x = beta * xp.cos(xp.pi * k / M)
     # Find the window's DFT coefficients
     # Use analytic definition of Chebyshev polynomial instead of expansion
     # from scipy.special. Using the expansion in scipy.special leads to errors.
-    p = np.zeros(x.shape)
-    p[x > 1] = np.cosh(order * np.arccosh(x[x > 1]))
-    p[x < -1] = (2 * (M % 2) - 1) * np.cosh(order * np.arccosh(-x[x < -1]))
-    p[np.abs(x) <= 1] = np.cos(order * np.arccos(x[np.abs(x) <= 1]))
+    p = xp.zeros(x.shape)
+    p[x > 1] = xp.cosh(order * xp.acosh(x[x > 1]))
+    p[x < -1] = (2 * (M % 2) - 1) * xp.cosh(order * xp.acosh(-x[x < -1]))
+    p[abs(x) <= 1] = xp.cos(order * xp.acos(x[abs(x) <= 1]))
 
     # Appropriate IDFT and filling up
     # depending on even/odd M
     if M % 2:
-        w = np.real(sp_fft.fft(p))
+        w = xp.real(xp.fft.fft(p))
         n = (M + 1) // 2
         w = w[:n]
-        w = np.concatenate((w[n - 1:0:-1], w))
+        w = xp.concat((w[n - 1:0:-1], w))
     else:
-        p = p * np.exp(1.j * np.pi / M * np.r_[0:M])
-        w = np.real(sp_fft.fft(p))
+        p = p * xp.exp(1.j * xp.pi / M * xp.arange(M))
+        w = xp.real(xp.fft.fft(p))
```

Comment:
I think we should stick with `sp_fft` here, as that is array API compatible. As written, this is not fully portable, since `xp.fft` is only an optional extension.

Reply:
Done